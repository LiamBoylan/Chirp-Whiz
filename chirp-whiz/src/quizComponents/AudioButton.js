import React from 'react';

/*
  Simple audio button used in many features in the app. Prop is a link to
  the sound clip that's meant to be played. Looks different depending on the
  browser. Meant to play bird audio.
*/

function AudioButton(props) {
      return (
        <div data-testid='audioButton'>
          <audio controls id="birdQAudio">
            <source src={props.sound} type="audio/mpeg"></source>
          </audio>
        </div>
      );
}

export default AudioButton;