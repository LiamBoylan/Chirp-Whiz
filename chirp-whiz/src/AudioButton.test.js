import React from 'react';
import ReactDOM from 'react-dom';
import AudioButton from './quizComponents/AudioButton';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'

// Integration tests

it('AudioButton snapshot', () => {
    const container = render(<AudioButton />)
    expect(container.firstChild).toMatchSnapshot();
});

it("renders without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<AudioButton />,div);
});