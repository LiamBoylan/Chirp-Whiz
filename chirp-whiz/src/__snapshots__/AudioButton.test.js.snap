// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioButton snapshot 1`] = `undefined`;
